File: android/lca/src/main/java/com/dreampany/lca/data/source/remote/CoinMarketCapGraphService.java
package com.dreampany.lca.data.source.remote;

import com.dreampany.lca.api.cmc.model.CmcGraph;

import io.reactivex.Maybe;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Created by Hawladar Roman on 29/5/18.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 */
public interface CoinMarketCapGraphService {

    @GET("currencies/{slug}/{startTime}/{endTime}/")
    Call<CmcGraph> getGraph(@Path("slug") String slug, @Path("startTime") long startTime, @Path("endTime") long endTime);

    //@Headers("Cache-Control: max-age=640000")
    @GET("currencies/{slug}/{startTime}/{endTime}/")
    Maybe<CmcGraph> getGraphRx(@Path("slug") String slug, @Path("startTime") long startTime, @Path("endTime") long endTime);
}

File: android/crypto/src/main/java/com/dreampany/crypto/api/crypto/remote/service/CoinMarketCapGraphService.kt
package com.dreampany.crypto.api.crypto.remote.service

import com.dreampany.crypto.api.crypto.misc.Constants
import com.dreampany.crypto.api.crypto.model.cmc.CryptoGraph
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Created by roman on 10/11/20
 * Copyright (c) 2020 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
interface CoinMarketCapGraphService {
    @GET(Constants.Apis.CoinMarketCap.GRAPH)
    fun read(
        @Path(Constants.Keys.CMC.SLUG) slug: String,
        @Path(Constants.Keys.CMC.START_TIME) startTime: Long,
        @Path(Constants.Keys.CMC.END_TIME) endTime: Long
     ): Call<CryptoGraph>
}